code/game/machinery/bots/ed209bot.dm

```diff
@@ -572,7 +573,8 @@ Auto Patrol[]"},
 			else if(istype(W, /obj/item/clothing/suit/bluetag))
 				lasercolor = "b"
 			if(lasercolor || istype(W, /obj/item/clothing/suit/armor/vest))
-				user.drop_item()
+				if(!user.unEquip(W))
+					return
 				qdel(W)

```

Comment:
the `if(!user.unEquip(W))` check needs to be right after if(2), otherwise, if the check fails and lasercolor has been set to r or b then the next item to be used on the assembly would be qdel'd no matter what it is.